File: codingsample/202008/20200822/Island_of_Knowledge_4_avoidObstacles.java
package Island_of_Knowledge;

public class Island_of_Knowledge_4_avoidObstacles {
	static int avoidObstacles(int[] inputArray) {
		int len = inputArray.length;
		for(int i=2;i<1002;i++) {
			for(int j=0;j<len;j++) {
				if(inputArray[j]%i==0) break;
				else if(inputArray[j]%i!=0&&j==len-1) return i;
			}
		}
		return 1;
	}

	public static void main(String[] args) {
		int[] a = {5,3,6,7,9};
		System.out.println(avoidObstacles(a));

	}

}

File: etc/~2020.11.26 - codingsample/202008/20200822/Island_of_Knowledge_4_avoidObstacles.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h> 

 typedef struct arr_integer {
   int size;
   int *arr;
 } arr_integer;

 arr_integer alloc_arr_integer(int len) {
   arr_integer a = {len, len > 0 ? malloc(sizeof(int) * len) : NULL};
   return a;
 }
arr_integer insertEM(arr_integer a, int *b){
	int len = a.size,i;
	for(i=0;i<len;i++) a.arr[i] = b[i];
	return a;
}

int avoidObstacles(arr_integer inputArray) {
	int i,j;
	for(i=2;i<1002;i++){
		for(j=0;j<inputArray.size;j++){
			if(inputArray.arr[j]%i==0) break;
			else if(j==inputArray.size-1&&inputArray.arr[j]%i!=0) return i;
		}
	}
	return 1;
}


int main() {
	arr_integer input = alloc_arr_integer(5);
	int a[5] = {5,3,6,7,9};
	insertEM(input,a);
	printf("%d",avoidObstacles(input));
	
}


